### DISCUSSION TITLE
Failed to load plugin remotely-save

### DISCUSSION THREAD

**Discussion by rk-main** [remotely-save/remotely-save#124] [Q&A]
I have a windows 11 machine and an obsidian vault inside a Cryptomator vault which is synced with a Dropbox. In Cryptomator, you can choose whether to use Dokany or WebDav as a file system. Obsidian does support Dokany out of the box, but since all other file operations are very slow, i would rather make Obsidian work with Webdav. So I wanted to try your Plugin, if it can do the job.

I can install it, but when I want to toggle the switch to enable it in my Obsidian settings, i get the error message:
"Failed to load plugin remotely-save".

Upon inspecting the console, I find this error message:

app.js:formatted:72211 Plugin failure: remotely-save SyntaxError: Unexpected token '}'
    at eval (<anonymous>)
    at app.js:formatted:72077
    at e.<anonymous> (app.js:formatted:72078)
    at app.js:formatted:11715
    at Object.next (app.js:formatted:11728)
    at s (app.js:formatted:11613)
(anonymous) @ app.js:formatted:72211
(anonymous) @ app.js:formatted:11715
(anonymous) @ app.js:formatted:11728
a @ app.js:formatted:11620
Promise.then (async)
l @ app.js:formatted:11631
(anonymous) @ app.js:formatted:11633
T @ app.js:formatted:11610
e.enablePlugin @ app.js:formatted:72186
(anonymous) @ app.js:formatted:72252
(anonymous) @ app.js:formatted:11715
(anonymous) @ app.js:formatted:11728
(anonymous) @ app.js:formatted:11633
T @ app.js:formatted:11610
e.enablePluginAndSave @ app.js:formatted:72248
(anonymous) @ app.js:formatted:97644
(anonymous) @ app.js:formatted:11715
(anonymous) @ app.js:formatted:11728
(anonymous) @ app.js:formatted:11633
T @ app.js:formatted:11610
(anonymous) @ app.js:formatted:97640
t.setValue @ app.js:formatted:52888
t.onClick @ app.js:formatted:52898

Inspecting the code at app.js.formatted:72211, i don't see why any '}' should be unexpected there. Do you have any clue what is happening?

Any help would be greatly appreciated.

cheers

Rainer from Germany

**Comment by fyears** [comment]
Could you please try using a **test** vault located on a regular disk location (instead of Cryptomator or any other “simulated file system” ), and see if the plugin could be enabled in the test vault?

**Reply by rk-main** [reply]
Hello fyears,

thank you a for your answer and picking up the issue! I created a Obsidian vault on regular disk (NTFS file system) and the I was able to install and switch on the plugin just fine. (I did not do any plug-in specific testing on the regular file system yet, to inspect if it really worked flawlessly. Please instruct me to do something like this, if you need further information. I was just able to toggle ON the switch of the Remotely Save plugin in the Obsidian Settings.)

For the Obsidian vault inside my cryptomator vault, I still have the same behaviour: If i mount the cryptomator vault with Dokany, I can access my Obsidian notes, but I cannot switch an the plugin (persistent error message like above). If I mount th cryptomator vault with Webdav filesystem, Obsidian throws the usual error: 
"An error occurred while loading Obsidian. Error: EISDIR: illegal operation on a directory, watch 'Z:\path\to\vault\'

Any more help is greatly appreciated.

Cheers

**Reply by fyears** [reply]
I am sorry, but as you said, the plugin can be turned on successfully on regular disk system, but not on cryptomator system. It's usually because that the simulated system behaves differently to regular system. I guess it's not something I could do here because I just call the api provided by the Obsidian. I suggest you post a bug report to Obsidian's bug list.

**Reply by rk-main** [reply]
Hello, fyears,

thanks again. I had the impression that this plugins special purpose was to facilitate obsidian to work and different filesystems / cloud storage locations. Or am I mistaken?

Cheers

**Reply by fyears** [reply]
This plugin is designed to work for local disk storage and remote cloud providers. For the local disk part, it relies on the apis provided by Obsidian to list and read files, not directly list and read files by itself magically. But for you cryptomator local disk, it seems that something goes wrong while **Obsidian** is reading and listing files.
